HPXMLtoOpenStudio/resources/hpxml.rb

```diff
@@ -3248,6 +3250,41 @@ def delete
 
     def check_for_errors
       errors = []
+
+      { 'Heating Season' => @seasons_heating_begin_month, 'Cooling Season' => @seasons_cooling_begin_month }.each do |season, begin_month|
+        next unless not begin_month.nil?
+
+        valid_months = (1..12).to_a
+        if not valid_months.include? begin_month
+          errors << "#{season} Begin Month (#{begin_month}) must be one of: #{valid_months.join(', ')}."
+        end
+      end
+
+      { 'Heating Season' => @seasons_heating_end_month, 'Cooling Season' => @seasons_cooling_end_month }.each do |season, end_month|
+        next unless not end_month.nil?
+
+        valid_months = (1..12).to_a
+        if not valid_months.include? end_month
+          errors << "#{season} End Month (#{end_month}) must be one of: #{valid_months.join(', ')}."
+        end
+      end
+
+      months_days = { [1, 3, 5, 7, 8, 10, 12] => (1..31).to_a, [4, 6, 9, 11] => (1..30).to_a, [2] => (1..28).to_a }
+      months_days.each do |months, valid_days|
+        { 'Heating Season' => [@seasons_heating_begin_month, @seasons_heating_begin_day, @seasons_heating_end_month, @seasons_heating_end_day], 'Cooling Season' => [@seasons_cooling_begin_month, @seasons_cooling_begin_day, @seasons_cooling_end_month, @seasons_cooling_end_day] }.each do |season, months_and_days|
+          begin_month, begin_day, end_month, end_day = months_and_days
+          if (not begin_day.nil?) && (months.include? begin_month)
+            if not valid_days.include? begin_day
+              errors << "#{sim_ctl} Begin Day of Month (#{begin_day}) must be one of: #{valid_days.join(', ')}."
+            end
+          end
+          next unless (not end_day.nil?) && (months.include? end_month)
+          if not valid_days.include? end_day
+            errors << "#{sim_ctl} End Day of Month (#{end_day}) must be one of: #{valid_days.join(', ')}."
+          end
+        end
+      end
```

Comment:
Don't we already have similar code for other dates? Please create a single method that performs this kind of logic.

Reply:
Add the check that the heating + cooling seasons span the year to here. (You can leave the error-checking in hvac.rb as a backup.) And change the cryptic error message from "Must be in either heating or cooling season" to "HeatingSeason and CoolingSeason, when combined, must span the entire year."

Reply:
Oops, just realized the error check can't be here because it needs to use both the heating_season and cooling_season objects. So move to [here](https://github.com/NREL/OpenStudio-HPXML/blob/master/HPXMLtoOpenStudio/resources/hpxml.rb#L5476) instead.